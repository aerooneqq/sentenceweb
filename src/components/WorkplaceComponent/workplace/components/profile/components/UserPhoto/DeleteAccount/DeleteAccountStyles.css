#deleteAccountBtn{
  width: 150px;
  height: 40px;

  margin-top: 20px;

  background: red;

  border-width: 1.5px;
  border-style: solid;
  border-color: red;
  border-radius: 5px;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

#deleteAccountBtn:hover{
  background: linear-gradient(23deg, #FF5F5F, red);
}

#deleteAccountBtn:focus{
  outline: none;
}

#deleteAccBtnText {
  font-family: Arial;
  font-size: 15px;
  color: white;

  margin-left: 5px;
}
