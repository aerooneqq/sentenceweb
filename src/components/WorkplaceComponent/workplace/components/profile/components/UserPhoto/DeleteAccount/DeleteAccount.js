import React, {Component} from "react"

//Styles
import "./DeleteAccountStyles.css"

//Icons
import deleteAccountIconStatic from "./img/delete_account_icon_static.png";
import deleteAccountIconActive from "./img/delete_account_icon_active.png";
import UserService from "../../../../../../../../services/UserServices/UserService";
import { alertAppMessage } from "../../../../../../../ApplicationMessage/ApplicationMessageManager";
import ResponseService from "../../../../../../../../services/ResponseService/ReponseService";

export default class DeleteAccount extends Component {

  constructor(props) { 
    super(props);

    this.state = { 
      isHovered: false
    };

    this.handleMouseLeaveEnter = this.handleMouseLeaveEnter.bind(this);
    this.deleteAccount = this.deleteAccount.bind(this);
  }

  handleMouseLeaveEnter() { 
    this.setState(prevState => { 
      return { 
        isHovered: !prevState.isHovered
      }
    })
  }

  deleteAccount() {
    new UserService(localStorage.getItem("token")).deleteAccount().then(() => { 
      alertAppMessage("The account was deleted!", "success");
      localStorage.removeItem("token");
    }).catch(er => { 
      new ResponseService().alertErrorMessage(er, "The error happened while deleting the account");
    })
  }

  render() {
    return (
      <div className = "toolTipContainer"
           onMouseEnter = {this.handleMouseLeaveEnter}
           onMouseLeave = {this.handleMouseLeaveEnter}>
        <button id = "deleteAccountBtn" onClick = {this.deleteAccount}>
          <img id = "deleteAccountIcon" 
               src = {this.state.isHovered === true ?  deleteAccountIconActive : deleteAccountIconStatic}
               alt = "Delete" />
        </button>
        <span className = "toolTipText">Delete account</span>
      </div>
    )
  }
}
