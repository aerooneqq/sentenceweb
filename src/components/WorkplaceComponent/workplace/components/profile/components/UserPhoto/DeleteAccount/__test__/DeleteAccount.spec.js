import React from 'react'
import { shallow, mount, render } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import DeleteAccount from "../DeleteAccount";

describe("ProfileDataItem", () => { 
    it("should render corrctly", () => { 
        const deleteAccount = shallow(<DeleteAccount />);

        expect(shallowToJson(deleteAccount)).toMatchSnapshot();
    });

    it("should call delete account method", () => { 
        const deleteAccount = shallow(<DeleteAccount />);

        const spy = jest.spyOn(deleteAccount.instance(), 'deleteAccount').mockImplementation(() => {});
        deleteAccount.find('button').simulate('click');
        
        expect(spy).toBeCalled()
    });
});