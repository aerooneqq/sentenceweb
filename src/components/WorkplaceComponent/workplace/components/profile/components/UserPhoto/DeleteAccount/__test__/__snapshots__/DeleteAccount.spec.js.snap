// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDataItem should render corrctly 1`] = `
<div
  className="toolTipContainer"
>
  <button
    id="deleteAccountBtn"
    onClick={[Function]}
  >
    <img
      alt="Delete"
      id="deleteAccountIcon"
      src="delete_account_icon.svg"
    />
  </button>
  <span
    className="toolTipText"
  >
    Delete account
  </span>
</div>
`;
